### PR TITLE
Allow free policies to have their default rooms be shown

### PR DESCRIPTION
**Pull request by yuwenmemon**
@MonilBhavsar please review

### Details
Branching off a revert of https://github.com/Expensify/App/pull/9460, adjust the logic so that we only show default rooms that belong to free plan workspaces, instead of showing all policy rooms to anyone who has a free plan workspace.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/214193

### Tests
1. change [this line](https://github.com/Expensify/App/blob/df58956f8fb1992378fa5c64150916675073d4b2/src/libs/Permissions.js#L11) to `return false`
1. Log into an account on dev, create some Free Plan policy types. To do this go to the green plus on the bottom left and create a Workspace.
1. After it's created verify that after creating your workspace you see the default rooms for the workspace show up in the LHN: https://user-images.githubusercontent.com/4741899/173680577-10d77f0f-c3ff-4401-8ff7-01e86d42a281.mp4
1. **Verify** that you cannot find the domain room for your domain that you're logged into Screen Shot 2022-06-14 at 4 20 10 PM
1. Log in via another account that does not belong to a free plan policy (workspace), but belongs to other policies.
1. **Verify** that you cannot see any default rooms (policy or domain): Screen Shot 2022-06-14 at 4 22 33 PM

1. Change that original line back, log in via **both accounts**, **Verify** you should be able to see all rooms now (including the domain room)

- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
1. Log into an account, create some Free Plan policy types. To do this go to the green plus on the bottom left and create a Workspace.
1. After it's created verify that after creating your workspace you see the default rooms for the workspace show up in the LHN: https://user-images.githubusercontent.com/4741899/173680577-10d77f0f-c3ff-4401-8ff7-01e86d42a281.mp4
1. **Verify** that you cannot find the domain room for your domain that you're logged into.
1. Log in via another account that does not belong to a free plan policy (workspace), but belongs to other policies.
1. **Verify** that you cannot see any default rooms (policy or domain.

- [ ] Verify that no errors appear in the JS console

### CODE REVIEW

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -441,14 +440,7 @@ function getOptions(reports, personalDetails, activeReportID, {
...
 
-        // If one is a member of a free policy, then they are allowed to see the Policy default rooms.
-        // For everyone else, one must be on the beta to see a default room.
-        const isMemberOfFreePolicy = PolicyUtils.isMemberOfFreePolicy(policies);
-        if (isMemberOfFreePolicy && !Permissions.canUseDefaultRooms(betas)) {
-            if (ReportUtils.isDomainRoom(report)) {
-                return;
-            }
-        } else if (ReportUtils.isDefaultRoom(report) && !Permissions.canUseDefaultRooms(betas)) {
+        if (ReportUtils.isDefaultRoom(report) && !Permissions.canUseDefaultRooms(betas) && ReportUtils.getPolicyType(report, policies) !== CONST.POLICY.TYPE.FREE) {
```
Comment (MonilBhavsar) [inline src/libs/OptionsListUtils.js:443]: NAB, It would be good to have a comment here.

File: src/pages/home/ReportScreen.js [hunk 2]
```diff
@@ -195,14 +194,9 @@ class ReportScreen extends React.Component {
...
 
-        // If one is a member of a free policy, then they are allowed to see the Policy default rooms.
-        // For everyone else, one must be on the beta to see a default room.
-        const isMemberOfFreePolicy = PolicyUtils.isMemberOfFreePolicy(this.props.policies);
-        if (isMemberOfFreePolicy && !Permissions.canUseDefaultRooms(this.props.betas)) {
-            if (ReportUtils.isDomainRoom(this.props.report)) {
-                return null;
-            }
-        } else if (ReportUtils.isDefaultRoom(this.props.report) && !Permissions.canUseDefaultRooms(this.props.betas)) {
+        if (!Permissions.canUseDefaultRooms(this.props.betas)
```
Comment (MonilBhavsar) [inline src/pages/home/ReportScreen.js:197]: Same here, can have a comment

**Review by MonilBhavsar — DISMISSED** [review]
Looks good and works fine!

**Comment by yuwenmemon** [comment]
Good point - added!

**Review by MonilBhavsar — APPROVED** [review]

**Comment by trjExpensify** [comment]
@MonilBhavsar - should this be working on NewDot staging because it was CP'd as per [the above](https://github.com/Expensify/App/pull/9632#issuecomment-1170738209)? 🤔 Here’s what I’ve tested: 

**Prerequisites:** 
- Don’t use an Expensify related email address to test this, as that will be on the defaultRooms beta with access to them on all policy types E.g expensifail.com, expensicorp.com etc 
- Test on staging, use incognito

**Member of a paid plan with no existing workspaces.** 

- Create and validate a new account on OldDot
- Create a collect or control policy
- Navigate across to NewDot staging and sign-in to that account
- Verify that there are no default rooms in the LHN list
- Refresh the page to double check they don’t appear

Result: PASS

image

**Member of a paid policy that creates a workspace**

- Now click the FAB and ’New Workspace`
- Verify that the defaultRooms for the workspace show up in the LHN
- Verify that the defaultRooms for the paid plan _do not_ show up in the LHN
- Refresh the page to double check they don’t appear

Result: FAIL. The paid plan defaultRooms appear after creating the workspace (and don't disappear after a refresh)

image

**Comment by trjExpensify** [comment]
Also tested the reverse for completeness with the same result. 

**Existing admin of a workspace, creates a new paid plan**

1. Sign up for a new account and validate it on OldDot
2. Navigate to NewDot staging and click the FAB and New workspace
3. Verify that the defaultRooms for the workspace show up in the LHN
4. Navigate to OldDot and create a paid plan
5. Navigate back to NewDot
6. Verify that the defaultRooms for the paid plan do not show up in the LHN

Result: FAIL

image

**Comment by trjExpensify** [comment]
>This isn't on staging yet. Looks like we're having some issues with Github actions. I'll ask in the slack here https://expensify.slack.com/archives/C07J32337/p1656578983148269

Ahhhhh. 👌

**Pull request merged**